subsys/bluetooth/host/settings.c

```diff
@@ -89,20 +89,15 @@ int bt_settings_decode_key(char *key, bt_addr_le_t *addr)
 static int set(int argc, char **argv, void *value_ctx)
 {
 	int len;
+	const struct bt_settings_handler *h;
 
-	if (argc > 1) {
-		const struct bt_settings_handler *h;
-
-		for (h = _bt_settings_start; h < _bt_settings_end; h++) {
-			if (!strcmp(argv[0], h->name)) {
-				argc--;
-				argv++;
+	for (h = _bt_settings_start; h < _bt_settings_end; h++) {
+		if (!strcmp(argv[0], h->name)) {
+			argc--;
+			argv++;
```

Comment:
I guess this change to the generic bt settings code should be split into its own commit with an explanation of what/why.

Reply:
Alright, it seems the top level (bt/) was restricted to just keys settings so this is changing to allow settings also only on top level.

Reply:
Actually my last comment is not quite right, the problem seems to be that we always expect an extra argument when in fact there could be settings that don't require one.

Reply:
It is now a separate patch.